Description: Add build support for GNU/Hurd and GNU/kFreeBSD
Author: Vasudev Kamath <vasudev@copyninja.info>
 Last-Update: 2016-12-28

--- a/mk/re.mk
+++ b/mk/re.mk
@@ -239,6 +239,16 @@
 	AR		:= ar
 	AFLAGS		:= cru
 endif
+ifeq ($(OS),gnu)
+	CFLAGS		+= -fPIC -DGNU
+	LIBS		+= -ldl
+	LFLAGS		+= -fPIC
+	SH_LFLAGS	+= -shared
+	MOD_LFLAGS	+=
+	APP_LFLAGS	+= -rdynamic
+	AR		:= ar
+	AFLAGS		:= cru
+endif
 ifeq ($(OS),darwin)
 	CFLAGS		+= -fPIC -dynamic -DDARWIN
 ifneq (,$(findstring Apple, $(CC_LONGVER)))
@@ -277,6 +287,16 @@
 	LFLAGS		+= -fPIC
 	SH_LFLAGS	+= -shared
 	MOD_LFLAGS	+=
+	APP_LFLAGS	+= -rdynamic
+	AR		:= ar
+	AFLAGS		:= cru
+	HAVE_KQUEUE	:= 1
+endif
+ifeq ($(OS),gnu/kfreebsd)
+	CFLAGS		+= -fPIC -DKFREEBSD -D_GNU_SOURCE
+	LFLAGS		+= -fPIC
+	SH_LFLAGS	+= -shared
+	MOD_LFLAGS	+=
 	APP_LFLAGS	+= -rdynamic
 	AR		:= ar
 	AFLAGS		:= cru
